### PR TITLE
refactored client configuration errors

### PR DESCRIPTION
**Pull request by ghubertpalo**
## Content
This PR features a rework of Configuration errors.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [X] Commit sequence broadly makes sense
  - [X] Key commits have useful messages
- PR
  - [X] No clippy warnings in the CI
  - [X] Self-reviewed the diff
  - [X] Useful pull request description
  - [X] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Comments
None
## Issue(s)

Relates to #798

### CODE REVIEW

**Review by dlachaumepalo — APPROVED** [review]

File: mithril-signer/src/configuration.rs [hunk 1]
```diff
@@ -76,34 +77,39 @@ pub struct Configuration {
 
 impl Configuration {
     /// Return the CardanoNetwork value from the configuration.
-    pub fn get_network(&self) -> Result<CardanoNetwork, ConfigError> {
+    pub fn get_network(&self) -> StdResult<CardanoNetwork> {
         CardanoNetwork::from_code(self.network.clone(), self.network_magic)
-            .map_err(|e| ConfigError::Message(e.to_string()))
+            .with_context(|| "Could not read Network from configuration.")
```
Comment (jpraynaud) [inline mithril-signer/src/configuration.rs:82]: The network name and magic would be useful in the context message

File: mithril-signer/src/configuration.rs [hunk 2]
```diff
@@ -76,34 +77,39 @@ pub struct Configuration {
...
         chain_observer: Arc<dyn ChainObserver>,
     ) -> StdResult<Arc<dyn EraReaderAdapter>> {
-        Ok(EraReaderAdapterBuilder::new(
+        EraReaderAdapterBuilder::new(
             &self.era_reader_adapter_type,
             &self.era_reader_adapter_params,
         )
         .build(chain_observer)
-        .map_err(|e| ConfigError::Message(format!("build era adapter failed {e}")))?)
+        .with_context(|| "Configuration: can not create era adapter.".to_string())
```
Comment (jpraynaud) [inline mithril-signer/src/configuration.rs:112]: The adapter type and params would be useful in the context message

File: mithril-signer/src/configuration.rs (hunk 1 above)
Comment (ghubertpalo) [reply mithril-signer/src/configuration.rs:82]: my bad, I thought it did not implement `Display`, but it does!

File: mithril-signer/src/configuration.rs (hunk 2 above)
Comment (ghubertpalo) [reply mithril-signer/src/configuration.rs:112]: added `Display` for `EraAdapterType`.

**Pull request merged**